packages/search/src/content/index.ts

```diff
@@ -89,13 +89,195 @@ export class ContentSearchService
  * A standalone function for searching content across the Portal API only or Portal API and the
  * Hub Indexer V3 API.
  *
- * @param request - the IContentSearchRequest instance for searching
+ * The `filter` argument allows the caller to filter results based on attributes exposed by AGO and
+ * the Hub API. A complete list of attributes can be found at the docs for `IContentSearchFilter`,
+ * but some examples include:
+ * - access
+ * - id
+ * - group
+ * - orgid
+ *
+ * The `options` argument allows the caller to specify more general attributes about the request itself.
+ * A complete list of options can be found at the docs for `IContentSearchFilter`, but some examples include:
+ * - the authenticated user
+ * - whether the call is happening in an enterprise environment
+ * - what site catalog to search
+ *
+ * Combined, both filters and options allow you to create complex queries against AGO / Hub API.
+ * Here are examples of some common use cases:
  *
  * ```js
- * import { searchContent } from '@esri/hub-search'
+ * import { searchContent, IBooleanOperator } from '@esri/hub-search';
+ * ...
+ *
+ * /////////////////////////////////////////////////////////////////////
+ * // Search for all public and private content associated with a site
+ * /////////////////////////////////////////////////////////////////////
+ * const options: IContentSearchOptions = {
+ *    site: 'https://my-site.hub.arcgis.com',
+ *    portal: 'https://www.arcgis.com',
+ *    // Note: only private content visible to the authenticated user will be returned
+ *    authentication: new UserSession(...),
+ * }
+ * const searchResults = await searchContent({ options });
+ * ...
+ *
+ * ///////////////////////////////////////////////////////////
+ * // Search for all public content associated with a site
+ * ///////////////////////////////////////////////////////////
+ * const options: IContentSearchOptions = {
+ *    site: 'https://my-site.hub.arcgis.com',
+ *    portal: 'https://www.arcgis.com',
+ *    authentication: new UserSession(...),
+ * }
+ * const filter: IContentSearchFilter = {
+ *    access: 'public' // Can also be expressed as { value: ['public'] }
```

Comment:
I'm thinking this comment may be confusing without further context that we get later. I could see someone asking: does it do anything different than just `public`? Why is it an array?

Reply:
Yeah, I can see that. Do you think I should make more examples using this extended structure with boolean operators @andrewctate ?

Reply:
I think you've got a good example below. I'm just wondering if there's a reason to keep this comment?

Reply:
No, I just threw in there because someone could theoretically see this in the future once searchContent is used in the app, but you're right in that we have another example and someone who runs into this should be able to parse out the structure from these and other docs 👍